src/Core/Services/RestService.cs

```diff
@@ -395,6 +395,22 @@ public bool TryGetRestRouteFromConfig([NotNullWhen(true)] out string? configured
             return false;
         }
 
+        /// <summary>
+        /// Helper method to extract the configured base route
+        /// </summary>
+        public string GetBaseRouteFromConfig()
+        {
+            if (_runtimeConfigProvider.TryGetConfig(out RuntimeConfig? config)
+                && config is not null
+                && config.Runtime is not null
+                && config.Runtime.BaseRoute is not null)
+            {
+                return config.Runtime.BaseRoute;
```

Comment:
An empty base-route i.e. `"/"` is allowed. But in such a case it is not included in the url. We should only return base-route when base-route is of the form "/xyz". You can be assured that the base-route always starts with forward slash as we have a validation for that in place.

Reply:
Thanks for bringing up this edge case. I tested with just `/` as the base-route. The `Location` header is getting constructed with the right URI. 

The baseURL is constructed using `https://learn.microsoft.com/dotnet/api/microsoft.aspnetcore.http.extensions.urihelper.buildabsolute` and it takes care of dealing with such strings I suppose.

For my understanding, it seems like we shouldn't be allowing just `/` in the base-route. Could you please explain the thought process behind why we chose allow.

Reply:
This is because '//' still is valid in a url i.e. a url can contain en empty component.

Reply:
Looks like this needs to be fixed: https://github.com/Azure/data-api-builder/blob/adb2be7f6ea2726d29da284fc935dc11a3741885/src/Core/Services/RestService.cs#L366. We would want to strip off leading '/' from the `route` before this check is performed.

Reply:
@ayush3797 `GetRouteAfterPathBase(string route)` expects the `route` argument to not have a starting slash '/'

Can you clarify what you are suggesting should be fixed?

Reply:
That we need to strip off all the leading '/' at line 365 i.e. TrimStart('/').

Reply:
Added the logic to remove leading slashes

Reply:
just as a follow to this thread, the leading slashes should not be removed because even though `//` is a valid URL, it is not a valid dab URL -> should result in a bad request or 404 not found.

Reply:
The existing logic is retained as is. The logic for removing leading slashes are not removed.